### ISSUE TITLE
tnt datasets church demo result is bad.

### ISSUE THREAD

**Issue by qq297110281** [NVlabs/neuralangelo#107]
any one try the tnt datasets's church demo? the result (50k iters) is not good.
![snapshot-church-100](https://github.com/NVlabs/neuralangelo/assets/8348315/61526d94-422a-434b-a2e4-f3e48050cb70)
![snapshot-church-201](https://github.com/NVlabs/neuralangelo/assets/8348315/74e131bc-b413-4898-a167-b00a46d01a81)
![snapshot-church-302](https://github.com/NVlabs/neuralangelo/assets/8348315/b8ba17c3-a8cd-4f38-b082-bd401f09ae6e)

here is my train paramters:

Training with 1 GPUs.
Using random seed 0
Make folder logs/church
* checkpoint:
   * save_epoch: 9999999999
   * save_iter: 20000
   * save_latest_iter: 9999999999
   * save_period: 9999999999
   * strict_resume: True
* cudnn:
   * benchmark: True
   * deterministic: False
* data:
   * name: dummy
   * num_images: None
   * num_workers: 4
   * preload: True
   * readjust:
      * center: [0.0, 0.0, 0.0]
      * scale: 1.0
   * root: datasets/church
   * train:
      * batch_size: 2
      * image_size: [1086, 1960]
      * subset: None
   * type: projects.neuralangelo.data
   * use_multi_epoch_loader: True
   * val:
      * batch_size: 1
      * image_size: [300, 541]
      * max_viz_samples: 16
      * subset: 1
* image_save_iter: 9999999999
* inference_args:
* local_rank: 0
* logdir: logs/church
* logging_iter: 9999999999999
* max_epoch: 9999999999
* max_iter: 500000
* metrics_epoch: None
* metrics_iter: None
* model:
   * appear_embed:
      * dim: 8
      * enabled: False
   * background:
      * enabled: False
      * encoding:
         * levels: 10
         * type: fourier
      * encoding_view:
         * levels: 3
         * type: spherical
      * mlp:
         * activ: relu
         * activ_density: softplus
         * activ_density_params:
         * activ_params:
         * hidden_dim: 256
         * hidden_dim_rgb: 128
         * num_layers: 8
         * num_layers_rgb: 2
         * skip: [4]
         * skip_rgb: []
      * view_dep: True
      * white: False
   * object:
      * rgb:
         * encoding_view:
            * levels: 3
            * type: spherical
         * mlp:
            * activ: relu_
            * activ_params:
            * hidden_dim: 256
            * num_layers: 4
            * skip: []
            * weight_norm: True
         * mode: idr
      * s_var:
         * anneal_end: 0.1
         * init_val: 3.0
      * sdf:
         * encoding:
            * coarse2fine:
               * enabled: True
               * init_active_level: 8
               * step: 5000
            * hashgrid:
               * dict_size: 20
               * dim: 4
               * max_logres: 11
               * min_logres: 5
               * range: [-2, 2]
            * levels: 16
            * type: hashgrid
         * gradient:
            * mode: numerical
            * taps: 4
         * mlp:
            * activ: softplus
            * activ_params:
               * beta: 100
            * geometric_init: True
            * hidden_dim: 256
            * inside_out: True
            * num_layers: 1
            * out_bias: 0.5
            * skip: []
            * weight_norm: True
   * render:
      * num_sample_hierarchy: 4
      * num_samples:
         * background: 0
         * coarse: 64
         * fine: 16
      * rand_rays: 512
      * stratified: True
   * type: projects.neuralangelo.model
* nvtx_profile: False
* optim:
   * fused_opt: False
   * params:
      * lr: 0.001
      * weight_decay: 0.01
   * sched:
      * gamma: 10.0
      * iteration_mode: True
      * step_size: 9999999999
      * two_steps: [300000, 400000]
      * type: two_steps_with_warmup
      * warm_up_end: 5000
   * type: AdamW
* pretrained_weight: None
* source_filename: projects/neuralangelo/configs/custom/church.yaml
* speed_benchmark: False
* test_data:
   * name: dummy
   * num_workers: 0
   * test:
      * batch_size: 1
      * is_lmdb: False
      * roots: None
   * type: imaginaire.datasets.images
* timeout_period: 9999999
* trainer:
   * amp_config:
      * backoff_factor: 0.5
      * enabled: False
      * growth_factor: 2.0
      * growth_interval: 2000
      * init_scale: 65536.0
   * ddp_config:
      * find_unused_parameters: False
      * static_graph: True
   * depth_vis_scale: 0.5
   * ema_config:
      * beta: 0.9999
      * enabled: False
      * load_ema_checkpoint: False
      * start_iteration: 0
   * grad_accum_iter: 1
   * image_to_tensorboard: False
   * init:
      * gain: None
      * type: none
   * loss_weight:
      * curvature: 0.0005
      * eikonal: 0.1
      * render: 1.0
   * type: projects.neuralangelo.trainer
* validation_iter: 5000
* wandb_image_iter: 10000
* wandb_scalar_iter: 100
cudnn benchmark: True
cudnn deterministic: False
Setup trainer.
Using random seed 0
model parameter count: 53,029,160
Initialize model weights using type: none, gain: None
Using random seed 0
Allow TensorFloat32 operations on supported devices

**Comment by mli0603** [comment]
Hi @qq297110281 

The Church scene does not come with the poses due to data corruption from TnT. Therefore, we did not report Church scene results for reproducibility. Nonetheless, I also experimented with the Church scene and I have found that COLMAP simply cannot recover the correct trajectory due to similarities/ambiguities of the chairs. Many poses end up opposite of where cameras are looking at. I would inspect your colmap results before running the Church scene. 

Let us know if you have made progress!

**Comment by hlpan** [comment]
model:
appear_embed:
dim: 8
enabled: False


maybe you should enable this.

model:
appear_embed:
dim: 32
enabled: True
